pypesto/storage/hdf5.py

```diff
@@ -0,0 +1,57 @@
+"""Convenience functions for working with HDF5 files"""
+import h5py
+from typing import Collection
+from numbers import Number
+
+
+def write_string_array(f: h5py.Group,
+                       path: str,
+                       strings: Collection) -> None:
+    """
+    Write string array to hdf5
+
+    Arguments:
+        f: h5py.File
+        path: path of the dataset to create
+        strings: list of strings
+    """
+    dt = h5py.special_dtype(vlen=str)
+    dset = f.create_dataset(path, (len(strings),), dtype=dt)
+    dset[:] = [s.encode('utf8') for s in strings]
+    dset.file.flush()
+
+
+def write_float_array(f: h5py.Group,
+                      path: str,
+                      values: Collection[Number],
+                      dtype='f8') -> None:
+    """
+    Write float array to hdf5
+
+    Arguments:
+        f: h5py.File
+        path: path of the dataset to create
+        values: array to write
+        dtype: datatype
+    """
+    dset = f.create_dataset(path, (len(values),), dtype=dtype)
+    dset[:] = values
+    dset.flush()
```

Comment:
above it is `dset.file.flush()`. Is both correct?

Reply:
Do be honest I didn't notice. It is like that here: 
https://github.com/ICB-DCM/parPE/blob/develop/python/parpe/hdf5.py

Reply:
Maybe it all should be f.file.flush()

Reply:
I think they are best all be removed here. Would make more sense to flush only after all data of one iteration have been written.